### PR TITLE
LCD menu updates (RA-DEC homing) and stop parking on setting home position via the LCD

### PR DESCRIPTION
**Pull request by Trigger-broom-289**
I've added RA and DEC homing to the LCD menu and disabled the auto tracking when setting up the home position (LCD).

### CODE REVIEW

**Comment by Trigger-broom-289** [comment]
Something did not work, will try again...

File: src/c65_startup.hpp [hunk 1]
```diff
@@ -33,8 +35,21 @@ enum startupState_t
         #define NO     2
         #define CANCEL 3
 
-startupState_t startupState = StartupIsInHomePosition;
-int isInHomePosition        = NO;
+        #if USE_HALL_SENSOR_RA_AUTOHOME == 1
```
Comment (ClutchplateDude) [inline src/c65_startup.hpp:38]: This won't even compile if someone has both Hall sensors.

File: src/c75_menuCTRL.hpp [hunk 2]
```diff
@@ -111,7 +115,98 @@ bool processControlKeys()
...
+                else if (key == btnDOWN)
+                {
+        #if USE_HALL_SENSOR_RA_AUTOHOME == 1
+                    ctrlState = HIGHLIGHT_RA_AUTO_HOME;
+        #else
+                    ctrlState = HIGHLIGHT_SERIAL;
+        #endif
+                }
+                //DEC-
+                else if (key == btnDOWN)
```
Comment (ClutchplateDude) [inline src/c75_menuCTRL.hpp:127]: This will never execute

File: src/c75_menuCTRL.hpp [hunk 3]
```diff
@@ -111,7 +115,98 @@ bool processControlKeys()
...
+                {
+                    ctrlState = HIGHLIGHT_MANUAL;
+                }
+                //DEC-
+                else if (key == btnDOWN)
+                {
+                    ctrlState = HIGHLIGHT_DEC_AUTO_HOME;
+                }
+                //-DEC
+                else if (key == btnDOWN)
```
Comment (ClutchplateDude) [inline src/c75_menuCTRL.hpp:170]: This also will never execute

File: src/c75_menuCTRL.hpp [hunk 4]
```diff
@@ -130,10 +235,28 @@ bool processControlKeys()
...
+                else if (key == btnUP)
+                {
+        #if USE_HALL_SENSOR_RA_AUTOHOME == 1
+                    ctrlState = HIGHLIGHT_RA_AUTO_HOME;
+        #else
+                    ctrlState = HIGHLIGHT_MANUAL;
+        #endif
+                }
+                //DEC-
+                else if (key == btnUP)
```
Comment (ClutchplateDude) [inline src/c75_menuCTRL.hpp:251]: Again, will never run

File: src/c65_startup.hpp (hunk 1 above)
Comment (Trigger-broom-289) [reply src/c65_startup.hpp:38]: It does compile, and I'm using it...


root@heidenrod-obs:/home/OAT_GitHub/firmware/OpenAstroTracker-Firmware# pio run -e mksgenlv21 -t upload
Processing mksgenlv21 (platform: atmelavr; board: ATmega2560; framework: arduino)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
pre_script_patch_debug.py: Found WInterrupts.c: /root/.platformio/packages/framework-arduino-avr-megacore/cores/MegaCore/WInterrupts.c
CONFIGURATION: https://docs.platformio.org/page/boards/atmelavr/ATmega2560.html
PLATFORM: Atmel AVR (4.0.1) > ATmega2560
HARDWARE: ATMEGA2560 16MHz, 8KB RAM, 256KB Flash
DEBUG: Current (avr-stub) External (avr-stub, simavr)
PACKAGES:
 - framework-arduino-avr-megacore @ 2.1.3
 - tool-avrdude @ 1.60300.200527 (6.3.0)
 - toolchain-atmelavr @ 1.70300.191015 (7.3.0)
pre_script_patch_debug.py: Patching /root/.platformio/packages/framework-arduino-avr-megacore/cores/MegaCore/WInterrupts.c
pre_script_patch_debug.py: Replacement path: /tmp/tmpzcqjnj7i_patched_WInterrupts.c
pre_script_patch_debug.py: Build path: /home/OAT_GitHub/firmware/OpenAstroTracker-Firmware/.pio/build/mksgenlv21/FrameworkArduino/WInterrupts.c
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 19 compatible libraries
Scanning dependencies...
Dependency Graph
|-- TinyGPSPlus @ 1.0.3
|-- TMCStepper @ 0.7.3
|-- AccelStepper @ 1.64.0
|-- LiquidCrystal @ 1.0.7
|-- LiquidTWI2 @ 1.2.7
|-- U8g2 @ 2.34.11
|-- avr-debugger @ 1.2.0
|-- InterruptStepper @ 0.0.2+sha.d4cc206
|-- EEPROM @ 2.0
|-- MappedDict
|-- Wire @ 1.0
|-- SPI @ 1.0
|-- SoftwareSerial @ 1.1
|-- Embedded Template Library @ 20.32.1
Building in release mode
Compiling .pio/build/mksgenlv21/src/Core.cpp.o
Compiling .pio/build/mksgenlv21/src/DayTime.cpp.o
Compiling .pio/build/mksgenlv21/src/Declination.cpp.o
Compiling .pio/build/mksgenlv21/src/EPROMStore.cpp.o
Compiling .pio/build/mksgenlv21/src/EndSwitches.cpp.o
Compiling .pio/build/mksgenlv21/src/Gyro.cpp.o
Compiling .pio/build/mksgenlv21/src/HallSensorHoming.cpp.o
Compiling .pio/build/mksgenlv21/src/Latitude.cpp.o
Compiling .pio/build/mksgenlv21/src/LcdButtons.cpp.o
Compiling .pio/build/mksgenlv21/src/LcdMenu.cpp.o
Compiling .pio/build/mksgenlv21/src/Longitude.cpp.o
Compiling .pio/build/mksgenlv21/src/MeadeCommandProcessor.cpp.o
Compiling .pio/build/mksgenlv21/src/Mount.cpp.o
Compiling .pio/build/mksgenlv21/src/Sidereal.cpp.o
Compiling .pio/build/mksgenlv21/src/Utility.cpp.o
Compiling .pio/build/mksgenlv21/src/WifiControl.cpp.o
Compiling /tmp/tmpzcqjnj7i_patched_WInterrupts.c.o
Archiving .pio/build/mksgenlv21/libFrameworkArduino.a
Indexing .pio/build/mksgenlv21/libFrameworkArduino.a
Linking .pio/build/mksgenlv21/firmware.elf
Checking size .pio/build/mksgenlv21/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [=======   ]  65.5% (used 5362 bytes from 8192 bytes)
Flash: [====      ]  40.9% (used 106728 bytes from 261120 bytes)
Building .pio/build/mksgenlv21/firmware.hex
Configuring upload protocol...
AVAILABLE: wiring
CURRENT: upload_protocol = wiring
Looking for upload port...
Auto-detected: /dev/ttyUSB0
Uploading .pio/build/mksgenlv21/firmware.hex

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.01s

avrdude: Device signature = 0x1e9801 (probably m2560)
avrdude: reading input file ".pio/build/mksgenlv21/firmware.hex"
avrdude: writing flash (106728 bytes):

Writing | ################################################## | 100% 14.87s

avrdude: 106728 bytes of flash written
avrdude: verifying flash memory against .pio/build/mksgenlv21/firmware.hex:
avrdude: load data flash data from input file .pio/build/mksgenlv21/firmware.hex:
avrdude: input file .pio/build/mksgenlv21/firmware.hex contains 106728 bytes
avrdude: reading on-chip flash data:

Reading | ################################################## | 100% 11.07s

avrdude: verifying ...
avrdude: 106728 bytes of flash verified

avrdude: safemode: Fuses OK (E:FF, H:D8, L:FF)

avrdude done.  Thank you.

=================================================================================== [SUCCESS] Took 48.97 seconds ===================================================================================

Environment    Status    Duration
-------------  --------  ------------
mksgenlv21     SUCCESS   00:00:48.971
==================================================================================== 1 succeeded in 00:00:48.971 ====================================================================================
root@heidenrod-obs:/home/OAT_GitHub/firmware/OpenAstroTracker-Firmware#

Comment (Trigger-broom-289) [reply src/c65_startup.hpp:38]: # cat Configuration_local.hpp | grep -i autohome
#define USE_HALL_SENSOR_RA_AUTOHOME 1
#define USE_HALL_SENSOR_DEC_AUTOHOME 1

File: src/c75_menuCTRL.hpp (hunk 2 above)
Comment (Trigger-broom-289) [reply src/c75_menuCTRL.hpp:127]: I have posted a video showing that it works...

File: src/c75_menuCTRL.hpp (hunk 3 above)
Comment (Trigger-broom-289) [reply src/c75_menuCTRL.hpp:170]: I have posted a video showing that it works...

File: src/c75_menuCTRL.hpp (hunk 4 above)
Comment (Trigger-broom-289) [reply src/c75_menuCTRL.hpp:251]: That one might actually be broken, I did say it in the Discord channel...

**Pull request closed without merging**